packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -237,7 +237,8 @@ class FlutterDevice {
             ipv6,
             disableServiceAuthCodes,
           );
-        } on Exception catch(e) {
+        } on Object catch(e) {
+          // startDevelopmentService can throw subclasses of both Error and Exception
```

Comment:
:( 

What errors are thrown?

Reply:
Ugh, forgot I changed this before landing last week. We originally threw a `StateError` [here](https://github.com/dart-lang/sdk/blob/master/pkg/dds/lib/src/dds_impl.dart#L32) but past me thought that wasn't a good idea and changed it.